### ISSUE TITLE
LG UltraFine monitor support for "real" brightness changes

### ISSUE THREAD

**Issue by mustdobetter** [ojotoxy/Brightness#9]
Firstly, thank you so much for developing this and making it publicly available. I have 2x LG UltraFine 5k displays hooked up to my MacBook Pro, and since purchasing them I've been frustrated on a daily basis that although you may adjust their brightnesses via the usual Mac controls, their internal sensors are not used for automatic adjustment. This means I am constantly adjusting them manually using the TouchBar of the MacBook - less than ideal!

Having searched for a long time for a solution to what must be a really common problem for owners of these Apple-approved external displays, I came to the conclusion that I'd have to build a utility myself that adjusted them automatically based on my MacBook's auto brightness. Until of course I found your utility that does just that. So thank you again for solving this problem - I can't tell you how much I appreciate it!

Ok, so onto the actual "issue" I'm raising. As per your project readme, you use two methods to change the brightness of external monitors: Real and Fake. It seems that although the UltraFine's have support for changing brightness via macOS default controls (e.g. using brightness keys on keyboard, and on TouchBar etc), your utility employs the "Fake" method of changing the brightness, as I can see the Real brightness sliders/levels do not change when the adjustments are made. Can you tell me if there's any reason the utility couldn't use the Real method for these monitors? Has it anything to do with the improvement you've described in the readme: "Make it compatible with apple thunderbolt displays"?

**Comment by deepai-org** [comment]
Yes - this could probably be done. But I don't have such a monitor to test with :)

I would definitely appreciate any insight into what APIs are supported on these monitors. 

Will definitely add this feature if it is possible.

**Comment by GoldenChaos** [comment]
I am in the same situation as @mustdobetter! Willing to test any support you try to add. As far as I know, it supports the regular brightness change method that "real" apple monitors use. Meaning, if you just considered the LG UltraFine 4K and 5K to be apple monitors, I bet it would just work with the existing "real" setting.

If there's a way for me to manually add the UltraFine myself to the list of "real" monitors, happy to test this out.

**Comment by scsskid** [comment]
any update?

**Comment by GoldenChaos** [comment]
Someone else beat this app to the punch, and I can confirm that it works perfectly:

https://github.com/fnesveda/ExternalDisplayBrightness

**Comment by OCJvanDijk** [comment]
If nobody minds, I would also like to plug my own project, because I think more people could benefit from it and it's hard enough to discover as it is: https://github.com/OCJvanDijk/Brightness-Sync

It's specially designed for UltraFine displays and works with automatic brightness based on your ambient light sensor.

**Comment by scsskid** [comment]
this is great OCJ, thanks for you effort! works perfect und very good readme
